addons/project/models/project_task.py

```diff
@@ -992,28 +985,38 @@ def _ensure_fields_are_accessible(self, fields, operation='read', check_group_us
         """
         assert operation in ('read', 'write'), 'Invalid operation'
         if fields and (not check_group_user or self.env.user._is_portal()) and not self.env.su:
-            unauthorized_fields = set(fields) - (self.SELF_READABLE_FIELDS if operation == 'read' else self.SELF_WRITABLE_FIELDS)
+            readable, writeable = self._portal_accessible_fields()
+            unauthorized_fields = set(fields) - (readable if operation == 'read' else writeable)
             if unauthorized_fields:
                 if operation == 'read':
                     error_message = _('You cannot read the following fields on tasks: %(field_list)s', field_list=unauthorized_fields)
                 else:
                     error_message = _('You cannot write on the following fields on tasks: %(field_list)s', field_list=unauthorized_fields)
                 raise AccessError(error_message)
 
+    @api.model
+    @tools.ormcache()
+    def _portal_accessible_fields(self) -> tuple[frozenset[str], frozenset[str]]:
+        """Readable and writable fields by portal users."""
+        readable = frozenset(self.SELF_READABLE_FIELDS)
+        writeable = frozenset(self.SELF_WRITABLE_FIELDS)
+        return readable | writeable, writeable
+
     def _has_field_access(self, field, operation):
         if not super()._has_field_access(field, operation):
             return False
         if not self.env.su and self.env.user._is_portal():
             # additional checks for portal users
+            readable, writeable = self._portal_accessible_fields()
             if operation == 'read':
-                return field.name in self.SELF_READABLE_FIELDS
+                return field.name in readable
             if operation == 'write':
-                return field.name in self.SELF_WRITABLE_FIELDS
+                return field.name in writeable
         return True
 
     def _determine_fields_to_fetch(self, field_names, ignore_when_in_cache=False):
         if not self.env.su and self.env.user._is_portal():
-            valid_names = self.SELF_READABLE_FIELDS
+            valid_names, _ = self._portal_accessible_fields()
```

Comment:
It will override the translation function `_`. 😬

Reply:
The code is removed in the next commits and it does not override the function but masks it locally (not an issue).